lib/_http_outgoing.js

```diff
@@ -90,11 +90,12 @@ exports.OutgoingMessage = OutgoingMessage;
 
 OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 
-  if (callback) {
-    if (typeof callback !== 'function')
-      throw new TypeError('callback must be a function');
+  if (typeof callback === 'function') {
     this.on('timeout', callback);
   }
+  else if (callback) {
+    throw new TypeError('callback must be a function');
```

Comment:
Right... At least the event handler code doesn't need this if block as it is done anyway in `on`, `once`, `{add,remove}listener` functions